Library/OcBootManagementLib/BootEntryInfo.c

```diff
@@ -459,6 +459,25 @@ OcGetBootEntryIcon (
     return Status;
   }
 
+  if (BootEntry->Type == OC_BOOT_EXTERNAL_OS && BootEntry->PathName != NULL) {
+    // Try to load the icon from the same path with appended .icns extension
```

Comment:
Please use EDK II-style comments with extra surrodunging `//` here and below, i.e.:

```C
//
// Something 
//
```